cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -871,8 +871,7 @@ private void deleteMeta(Table table, Id.Artifact artifactId, byte[] oldData) thr
     // delete the old jar file
 
     try {
-      new NamespacedImpersonator(artifactId.getNamespace().toEntityId(),
-                                 impersonator).impersonate(new Callable<Void>() {
+      new EntityImpersonator(artifactId.getNamespace().toEntityId(), impersonator).impersonate(new Callable<Void>() {
```

Comment:
We should be passing the entity directly. In this case ArtifactId and when the doAs is called on EntityImpersonator the owner admin will give whom to impersonate for the entity.